src/middlewared/middlewared/plugins/pool.py

```diff
@@ -300,6 +304,376 @@ def pool_extend(self, pool):
             pool['is_decrypted'] = True
         return pool
 
+    @accepts(Dict(
+        'pool_create',
+        Str('name', required=True),
+        Bool('encryption', default=False),
+        Str('deduplication', enum=[None, 'ON', 'VERIFY', 'OFF'], null=True),
+        Dict(
+            'topology',
+            List('data', items=[
+                Dict(
+                    'datavdevs',
+                    Str('type', enum=['RAIDZ1', 'RAIDZ2', 'RAIDZ3', 'MIRROR', 'STRIPE'], required=True),
+                    List('disks', items=[Str('disk')], required=True),
+                ),
+            ], required=True),
+            List('cache', items=[
+                Dict(
+                    'cachevdevs',
+                    Str('type', enum=['STRIPE'], required=True),
+                    List('disks', items=[Str('disk')], required=True),
+                ),
+            ]),
+            List('log', items=[
+                Dict(
+                    'logvdevs',
+                    Str('type', enum=['STRIPE', 'MIRROR'], required=True),
+                    List('disks', items=[Str('disk')], required=True),
+                ),
+            ]),
+            List('spares', items=[Str('disk')], default=[]),
```

Comment:
Default for `Dict` is `{}`, how about `[]` as default for `List`?

Reply:
We have tackled this in a previous PR, having a not explicit default creates all sorts of problems when checking for user provided data.